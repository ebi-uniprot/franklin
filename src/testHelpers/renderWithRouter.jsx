import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';

export default (
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {}
) => {
  // eslint-disable-next-line react/prop-types
  const Wrapper = ({ children }) => (
    <Router history={history}>{children}</Router>
  );
  return {
    ...render(ui, { wrapper: Wrapper }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history,
  };
};
