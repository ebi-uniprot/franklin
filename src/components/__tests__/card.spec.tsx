import { screen, fireEvent } from '@testing-library/react';

import Card from '../card';

import renderWithRouter from '../../testHelpers/renderWithRouter';

describe('Card component', () => {
  test('should render', () => {
    const { asFragment } = renderWithRouter(
      <Card title="Title">
        <span>Some content</span>
      </Card>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render card with subtitle', () => {
    const { asFragment } = renderWithRouter(
      <Card title="Title" subtitle="Subtitle">
        <span>Some content</span>
      </Card>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render card with links', () => {
    const links = [
      {
        name: 'link',
        link: 'example.com',
        color: 'red',
      },
    ];

    const { asFragment } = renderWithRouter(
      <Card title="Title" subtitle="Subtitle" links={links}>
        <span>Some content</span>
      </Card>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render card with onclick', () => {
    const onClickMock = jest.fn();
    const { asFragment } = renderWithRouter(
      <Card title="Title" subtitle="Subtitle" onClick={onClickMock}>
        <span>Some content</span>
      </Card>
    );
    fireEvent.click(screen.getByRole('button'));
    expect(onClickMock).toHaveBeenCalled();
    expect(asFragment()).toMatchSnapshot();
  });
});
