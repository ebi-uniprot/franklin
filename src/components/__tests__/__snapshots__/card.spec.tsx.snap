// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <section>
      <div
        class="card__header"
      >
        <h2
          class="card__title"
        >
          Title
        </h2>
      </div>
      <div
        class="card__content"
      >
        <span>
          Some content
        </span>
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Card component should render card with links 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <section>
      <div
        class="card__header"
      >
        <h2
          class="card__title"
        >
          Title
        </h2>
        <div
          class="card__subtitle"
        >
          Subtitle
        </div>
      </div>
      <div
        class="card__content"
      >
        <span>
          Some content
        </span>
      </div>
    </section>
    <section
      class="card__actions"
    >
      <span
        class="card__link"
        style="border-bottom: 0.125rem solid red;"
      >
        <a
          href="/example.com"
        >
          link
        </a>
      </span>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Card component should render card with onclick 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <section
      class="card--has-hover"
      role="button"
      tabindex="0"
    >
      <div
        class="card__header"
      >
        <h2
          class="card__title"
        >
          Title
        </h2>
        <div
          class="card__subtitle"
        >
          Subtitle
        </div>
      </div>
      <div
        class="card__content"
      >
        <span>
          Some content
        </span>
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Card component should render card with subtitle 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <section>
      <div
        class="card__header"
      >
        <h2
          class="card__title"
        >
          Title
        </h2>
        <div
          class="card__subtitle"
        >
          Subtitle
        </div>
      </div>
      <div
        class="card__content"
      >
        <span>
          Some content
        </span>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
